src/main/java/uk/gov/hmcts/darts/arm/service/impl/UnstructuredToArmBatchProcessorImpl.java

```diff
@@ -85,31 +88,45 @@ private void createAndSendBatchFile(List<ExternalObjectDirectoryEntity> eodsForB
-                    armEod = currentEod;
-                    batchItem.setArmEod(armEod);
-                    updateArmEodToArmIngestionStatus(currentEod, batchItem, batchItems, archiveRecordsFile, userAccount);
-                } else {
-                    armEod = unstructuredToArmHelper.createArmEodWithArmIngestionStatus(currentEod, batchItem, batchItems, archiveRecordsFile, userAccount);
-                }
-
-                String rawFilename = unstructuredToArmHelper.generateRawFilename(armEod);
 
-                if (shouldPushRawDataToArm(batchItem)) {
-                    pushRawDataAndCreateArchiveRecordIfSuccess(batchItem, rawFilename, userAccount);
-                } else if (shouldAddEntryToManifestFile(batchItem)) {
-                    batchItem.setArchiveRecord(archiveRecordService.generateArchiveRecordInfo(batchItem.getArmEod().getId(), rawFilename));
+        List<Callable<Void>> tasks = eodsForBatch
+            .stream()
+            .map(currentEod -> (Callable<Void>) () -> {
+                var batchItem = new ArmBatchItem();
+                try {
+                    ExternalObjectDirectoryEntity armEod;
+                    if (isEqual(currentEod.getExternalLocationType(), EodHelper.armLocation())) {
+                        //retry existing attempt that has previously failed.
+                        armEod = currentEod;
+                        batchItem.setArmEod(armEod);
+                        updateArmEodToArmIngestionStatus(currentEod, batchItem, batchItems, archiveRecordsFile, userAccount);
+                    } else {
+                        armEod = unstructuredToArmHelper.createArmEodWithArmIngestionStatus(currentEod, batchItem, batchItems, archiveRecordsFile, userAccount);
+                    }
+
+                    String rawFilename = unstructuredToArmHelper.generateRawFilename(armEod);
+
+                    if (shouldPushRawDataToArm(batchItem)) {
+                        pushRawDataAndCreateArchiveRecordIfSuccess(batchItem, rawFilename, userAccount);
+                    } else if (shouldAddEntryToManifestFile(batchItem)) {
+                        batchItem.setArchiveRecord(archiveRecordService.generateArchiveRecordInfo(batchItem.getArmEod().getId(), rawFilename));
+                    }
+                } catch (Exception e) {
+                    log.error("Unable to batch push EOD {} to ARM", currentEod.getId(), e);
+                    recoverByUpdatingEodToFailedArmStatus(batchItem, userAccount);
                 }
-            } catch (Exception e) {
-                log.error("Unable to batch push EOD {} to ARM", currentEod.getId(), e);
-                recoverByUpdatingEodToFailedArmStatus(batchItem, userAccount);
+                return null;
+            })
+            .toList();
+
+        try {
+            AsyncUtil.invokeAllAwaitTermination(tasks, unstructuredToArmProcessorConfiguration.getThreads(), 2, TimeUnit.HOURS);
```

Comment:
This is a default time out so if the task takes longer then 2 hours then the threads will all be aborted. But it will auto close as soon as all threads are completed :)

I believe the batch size is due to increase and have lock times changed following this spike: https://tools.hmcts.net/jira/browse/DMP-4064. It would be worth changing this to 1.5h to match the max lock time defined